server/openslides/poll/views.py

```diff
@@ -246,18 +262,21 @@ def assert_can_vote(self, poll, request):
                 raise ValidationError("You can only vote on a started poll.")
 
             if not request.user.is_present or not in_some_groups(
-                request.user.id,
+                vote_user.id,
                 list(poll.groups.values_list("pk", flat=True)),
                 exact=True,
             ):
                 self.permission_denied(request)
 
             try:
-                self.add_user_to_voted_array(request.user, poll)
+                self.add_user_to_voted_array(vote_user, poll)
                 inform_changed_data(poll)
             except IntegrityError:
                 raise ValidationError({"detail": "You have already voted"})
 
+        if request.user != vote_user and request.user != vote_user.vote_delegated_to:
```

Comment:
can this check be moved up before the checking, if the user is present and in the right groups? Because than there must not be done any rollback by the DB since nothing was created.

Reply:
Yes, good catch